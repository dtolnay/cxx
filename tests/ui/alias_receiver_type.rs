#[cxx::bridge]
mod here {
    struct Shared {
        z: usize,
    }

    extern "C" {
        type C;
    }
}

// Rustfmt mangles the extern type alias.
// https://github.com/rust-lang/rustfmt/issues/4159
// ...normally would add #[rustfmt::skip], but that seems to interfere with the error spans.
#[cxx::bridge]
mod there {
    type Shared = crate::here::Shared;

    extern "C" {
        type C = crate::here::C;

        fn good(self: &C);
        fn bad(self: &Shared);
    }
}

fn main() {}
