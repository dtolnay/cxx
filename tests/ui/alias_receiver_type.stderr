error: shared type aliases cannot be used as a receiver type
  --> $DIR/alias_receiver_type.rs:23:22
   |
23 |         fn bad(self: &Shared);
   |                      ^^^^^^^
